### PR TITLE
[cc1352] add support for Texas Instruments CC1352

### PR DESCRIPTION
**Pull request by srickardti**
This Pull Request adds support for Texas Instruments CC1352 on the CC1352R1_LAUNCHXL development platform.

### CODE REVIEW

**Comment by srickardti** [comment]
I'm not sure what is making the Address Query test on the Posix simulation fail, but I don't think any of my changes are causing that.

File: examples/Makefile-cc1352 [hunk 1]
```diff
@@ -0,0 +1,293 @@
+#
+#  Copyright (c) 2017, The OpenThread Authors.
```
Comment (jwhui) [inline examples/Makefile-cc1352]: nit: "2018"? or "2017-2018"? (applies to other files below)

File: examples/platforms/cc1352/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+override CFLAGS                                      := $(filter-out -pedantic-errors,$(CFLAGS))
+override CXXFLAGS                                    := $(filter-out -pedantic-errors,$(CXXFLAGS))
+
+libopenthread_cc1352_a_CPPFLAGS                       = \
+    -I$(top_srcdir)/include                             \
+    -I$(top_srcdir)/src                                 \
+    -I$(top_srcdir)/src/core                            \
+    -I$(top_srcdir)/examples/platforms                  \
+    -I$(top_srcdir)/examples/platforms/cc1352           \
+    -I$(top_srcdir)/third_party/ti/devices/cc13x2_cc26x2 \
```
Comment (jwhui) [inline examples/platforms/cc1352/Makefile.am]: nit: align backslash

**Comment by srickardti** [comment]
@jwhui, updated the copyright notices to 2018 and lined up the makefiles. Thanks for looking at it.

**Pull request merged**